### ISSUE TITLE
appsync: release 1.91.0 breaks appsync resolvers

### ISSUE THREAD

**Issue by smacintyre** [aws/aws-cdk#13357]
Release 1.91.0 contains an breaking change for appsync revolvers and appears to make deploying unit resolvers impossible now. This was caused by the PR: #12973 -- this was not documented in the breaking changes section of the release notes.

Also the documentation is not up to date. [According to the docs](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-appsync.BaseResolverProps.html#pipelineconfigspan-classapi-icon-api-icon-experimental-titlethis-api-element-is-experimental-it-may-change-without-noticespan), if `pipelineConfig` is an empty array or undefined, a unit resolver is created. The change in #12973 breaks this.

Reverting to 1.90.1 doesn't fix the issue.

### Reproduction Steps

```ts
   const data = graphqlApi.addDynamoDbDataSource('MainDataSource', table);

    data.createResolver({
      typeName: 'Mutation',
      fieldName: 'createVideo',
      requestMappingTemplate: MappingTemplate.fromFile(
        join(this.resolverDir, 'Mutation.createVideo.req.vtl')),
      responseMappingTemplate: MappingTemplate.dynamoDbResultItem(),
    });
```

### What did you expect to happen?

To create a unit resolver -- this is what happened in prior versions. 

### What actually happened?

No resolver is created

### Environment

  - **CDK CLI Version  :*1.91.0*
  - **Framework Version: *1.91.0*
  - **Node.js Version: *v14.16.0*
  - **OS               :**
  - **Language (Version):*4.1.5*

### Other

Right now, I have no idea of how to work around this.

--- 

This is :bug: Bug Report

**Comment by cynicaljoy** [comment]
seems to be a duplicate of #13269

**Comment by MrArnoldPalmer** [comment]
The issue re: resolvers is in #13269. This was fixing a previously breaking change @smacintyre, I'm sorry this wasn't explicitly called out in the changelog I should have required the breaking change annotation on the PR but made a mistake.

This issue stems from the cloudformation/appsync side unfortunately and we aren't quite sure how to handle it from the CDK side for both problems. IE `only one resolver per field` and/or the "successful" deploy of resolvers that aren't wired correctly. Experimentation and investigation is required and contributions are welcome on this front.

As for workarounds, @bigkraig talks about versioning resolvers in his app in #13269 and looks like some other workarounds are mentioned in [this amplify issue](https://github.com/aws-amplify/amplify-cli/issues/682#issuecomment-611695534). Worst case scenario deleting and recreating the `GraphqlApi` may be required.

`@aws-cdk/aws-appsync` L2 constructs are experimental and more breaking changes are expected, we will do our best to note these in the changelog but their usage in production workflows is not recommended without extreme caution. In addition the docs need a refresh in places, I created a [separate issue](https://github.com/aws/aws-cdk/issues/13502) to track a general revision. Closing this in favor of #13269 & #13502.
